syntax = "proto3";

message FotoDePerfilDatos
{
    string idJugador = 1;
    bytes datos = 2;
}

message ActualizacionFotoDePerfil
{
    string idJugador = 1;
    string rutaImagenAntigua = 2;
    bytes datos = 3;
}

message FotoDePerfilRuta
{
    string rutaArchivo = 1;
}

service FotosDePerfil
{
    rpc SubirFotoDeUsuario(FotoDePerfilDatos) returns (FotoDePerfilRuta);
    rpc ObtenerFotoDePerfilUsuario(FotoDePerfilRuta) returns (FotoDePerfilDatos);
    rpc ActualizarFotoDePerfil(ActualizacionFotoDePerfil) returns (FotoDePerfilRuta);
}